File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        # optimal
        ans = [1]*len(nums)
        positiveIndex = 0
        negativeIndex = 1
        
        for i in range(len(nums)):
            if nums[i] < 0:
                ans[negativeIndex] = nums[i]
                negativeIndex += 2
            else:
                ans[positiveIndex] = nums[i]
                positiveIndex += 2
        return ans

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        //Brute Force
        int[] positive = new int[nums.length/2];
        int[] negative = new int[nums.length/2];
        int j = 0;
        int k = 0;
        for(int i = 0; i < nums.length;i++) {
            if(nums[i] < 0) {
                negative[j] = nums[i];
                j++;
            } else {
                positive[k] = nums[i];
                k++;
            }
        }
        
        for(int i = 0;i<nums.length/2;i++) {
            nums[2*i] = positive[i];
            nums[(2*i)+1] = negative[i];
        }
        return nums;
    }
}